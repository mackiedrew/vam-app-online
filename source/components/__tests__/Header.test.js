import Subject from "../Header";
jest.mock("../../containers/AddTrack", () => "AddTracks");

describe("<Header />", () => {
  it("renders without crashing", () => {
    shallow(<Subject />);
  });

  describe("buttons work", () => {
    it("Toggle Filters Button calls passed function", () => {
      const mockToggleFiltersMenu = sinon.spy();
      const subject = shallow(
        <Subject toggleFiltersMenu={mockToggleFiltersMenu} />
      );
      subject.find("button.toggle-filters").simulate("click");
      expect(mockToggleFiltersMenu.calledOnce).toBe(true);
    });
    it("Toggle Settings Button calls passed function", () => {
      const mockToggleSettingsMenu = sinon.spy();
      const subject = shallow(
        <Subject toggleSettingsMenu={mockToggleSettingsMenu} />
      );
      subject.find("button.toggle-settings").simulate("click");
      expect(mockToggleSettingsMenu.calledOnce).toBe(true);
    });
  });

  describe("renders correctly", () => {
    it("without props", () => {
      const tree = renderer.create(<Subject />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it("with props", () => {
      const tree = renderer
        .create(
          <Subject
            toggleFiltersMenu={sinon.spy()}
            toggleSettingsMenu={sinon.spy()}
          />
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
