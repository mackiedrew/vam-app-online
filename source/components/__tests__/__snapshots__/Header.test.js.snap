// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly with props 1`] = `
<header
  className="header"
>
  <div
    className="main-bar"
  >
    <button
      className="toggle-filters"
      onClick={[Function]}
    >
      <i
        className="material-icons icon"
        style={
          Object {
            "fontSize": "24px",
            "height": "24px",
            "width": "24px",
          }
        }
      >
        library_add
      </i>
    </button>
    <h1>
      VAM
    </h1>
    <button>
      <i
        className="material-icons icon"
        style={
          Object {
            "fontSize": "24px",
            "height": "24px",
            "width": "24px",
          }
        }
      >
        file_download
      </i>
    </button>
    <AddTracks />
    <button
      className="toggle-settings"
      onClick={[Function]}
    >
      <i
        className="material-icons icon"
        style={
          Object {
            "fontSize": "24px",
            "height": "24px",
            "width": "24px",
          }
        }
      >
        settings
      </i>
    </button>
  </div>
  <div
    className="tool-bar"
  >
    <button
      className="split-grain"
      onClick={[Function]}
    >
      <i
        className="material-icons icon"
        style={
          Object {
            "fontSize": "16px",
            "height": "16px",
            "width": "16px",
          }
        }
      >
        call_split
      </i>
    </button>
  </div>
</header>
`;

exports[`<Header /> renders correctly without props 1`] = `
<header
  className="header"
>
  <div
    className="main-bar"
  >
    <button
      className="toggle-filters"
      onClick={undefined}
    >
      <i
        className="material-icons icon"
        style={
          Object {
            "fontSize": "24px",
            "height": "24px",
            "width": "24px",
          }
        }
      >
        library_add
      </i>
    </button>
    <h1>
      VAM
    </h1>
    <button>
      <i
        className="material-icons icon"
        style={
          Object {
            "fontSize": "24px",
            "height": "24px",
            "width": "24px",
          }
        }
      >
        file_download
      </i>
    </button>
    <AddTracks />
    <button
      className="toggle-settings"
      onClick={undefined}
    >
      <i
        className="material-icons icon"
        style={
          Object {
            "fontSize": "24px",
            "height": "24px",
            "width": "24px",
          }
        }
      >
        settings
      </i>
    </button>
  </div>
  <div
    className="tool-bar"
  >
    <button
      className="split-grain"
      onClick={[Function]}
    >
      <i
        className="material-icons icon"
        style={
          Object {
            "fontSize": "16px",
            "height": "16px",
            "width": "16px",
          }
        }
      >
        call_split
      </i>
    </button>
  </div>
</header>
`;
